# **STORE SALES & PROFIT ANALYSIS**

## **PROJECT PURPOSE**

In the realm of uncertainty, the solution often lies within the exploration of your data. Every company's primary goal is not just to sell but to thrive and ensure sustained success through profitable operations. While some businesses face losses and others struggle to break even, the majority aim for profitability. However, achieving optimal profits requires companies to embrace data-driven decision-making.

Conducting real-time analysis of sales data holds the key to unraveling hidden insights, pinpointing areas for enhancement, and steering businesses toward increased sales and profitability. The future of success lies in leveraging the power of data.

Recognizing the importance of collaboration, I partnered with **Windfred Boakye-Kwarteng** to enhance the depth of this project. Together, we delve into the realms of sales and profit analysis, employing Python to unlock valuable insights and drive informed decision-making.

## **IMPORT LIBRARIES**

1. **pandas (import pandas as pd):** Pandas is a powerful data manipulation and analysis library in Python. It provides data structures like DataFrame and Series, which are essential for handling and analyzing structured data. With Pandas, you can easily load, manipulate, and analyze data, making it a fundamental library for data analysis and preparation.

2. **plotly.express (import plotly.express as px):** Plotly Express is a high-level data visualization library built on top of Plotly. It offers a simplified interface for creating a variety of interactive and visually appealing plots with minimal code. Plotly Express is particularly useful for creating charts like scatter plots, line charts, bar charts, and more, with a focus on ease of use.

3. **plotly.graph_objects (import plotly.graph_objects as go):** Plotly Graph Objects provides a lower-level interface compared to Plotly Express. It allows for more fine-grained control over the appearance and customization of plots. With this library, you can create sophisticated and customized visualizations using a broader set of configuration options.

4. **plotly.io (import plotly.io as pio):** Plotly IO is a module within Plotly that provides functionality for reading and writing different file formats for plots. It allows you to export plots to various formats such as HTML, JSON, static images, or interactive web-based visualizations.

5. **plotly.colors (import plotly.colors):** Plotly Colors provides a collection of predefined color scales and color-related functions. It is useful for customizing the color schemes of your plots, ensuring aesthetically pleasing and meaningful visualizations.


In [4]:


# These libraries collectively empower data scientists and analysts to efficiently handle data, 
# explore patterns, and communicate insights through visually compelling plots and charts.


import pandas as pd
import plotly.express as px
import plotly.graph_objects as go
import plotly.io as pio
import plotly.colors as colors


# A CHANGE TO REFLECT
# A CHANGE TO REFLECT


## **THE DATASET**


We had an ideal dataset for this task on Kaggle. For the task of Sales and Profit Analysis, we need to have a dataset with time-period data, sales data, product data, pricing data, customer data, etc.



The provided code below is using the pandas library to read data from a CSV (Comma-Separated Values) file named "StoreSaleData.csv." The pd.read_csv function is employed for this purpose, and the resulting DataFrame is stored in the variable named data. The encoding='latin-1' parameter specifies the character encoding of the CSV file, which is important for correctly interpreting the text in the file.

### **Metadata**
* Row ID => Unique ID for each row.
* Order ID => Unique Order ID for each Customer.
* Order Date => Order Date of the product.
* Ship Date => Shipping Date of the Product.
* Ship Mode=> Shipping Mode specified by the Customer.
* Customer ID => Unique ID to identify each Customer.
* Customer Name => Name of the Customer.
* Segment => The segment where the Customer belongs.
* Country => Country of residence of the Customer.
* City => City of residence of of the Customer.
* State => State of residence of the Customer.
* Postal Code => Postal Code of every Customer.
* Region => Region where the Customer belong.
* Product ID => Unique ID of the Product.
* Category => Category of the product ordered.
* Sub-Category => Sub-Category of the product ordered.
* Product Name => Name of the Product
* Sales => Sales of the Product.
* Quantity => Quantity of the Product.
* Discount => Discount provided.
* Profit => Profit/Loss incurred.

In [5]:


data = pd.read_csv("StoreSaleData.csv", encoding='latin-1')



In [6]:
# The head() method, by default, shows the first five rows of the DataFrame, 
# providing a quick overview of the data's structure and content.

data.head()


Row ID        Order ID  Order Date   Ship Date       Ship Mode Customer ID  \
0       1  CA-2016-152156   11/8/2016  11/11/2016    Second Class    CG-12520   
1       2  CA-2016-152156   11/8/2016  11/11/2016    Second Class    CG-12520   
2       3  CA-2016-138688   6/12/2016   6/16/2016    Second Class    DV-13045   
3       4  US-2015-108966  10/11/2015  10/18/2015  Standard Class    SO-20335   
4       5  US-2015-108966  10/11/2015  10/18/2015  Standard Class    SO-20335   

     Customer Name    Segment        Country             City  ...  \
0      Claire Gute   Consumer  United States        Henderson  ...   
1      Claire Gute   Consumer  United States        Henderson  ...   
2  Darrin Van Huff  Corporate  United States      Los Angeles  ...   
3   Sean O'Donnell   Consumer  United States  Fort Lauderdale  ...   
4   Sean O'Donnell   Consumer  United States  Fort Lauderdale  ...   

  Postal Code  Region       Product ID         Category Sub-Category  \
0       42420   South  FUR-BO-10001798        Furniture    Bookcases   
1       42420   South  FUR-CH-10000454        Furniture       Chairs   
2       90036    West  OFF-LA-10000240  Office Supplies       Labels   
3       33311   South  FUR-TA-10000577        Furniture       Tables   
4       33311   South  OFF-ST-10000760  Office Supplies      Storage   

                                        Product Name     Sales  Quantity  \
0                  Bush Somerset Collection Bookcase  261.9600         2   
1  Hon Deluxe Fabric Upholstered Stacking Chairs,...  731.9400         3   
2  Self-Adhesive Address Labels for Typewriters b...   14.6200         2   
3      Bretford CR4500 Series Slim Rectangular Table  957.5775         5   
4                     Eldon Fold 'N Roll Cart System   22.3680         2   

   Discount    Profit  
0      0.00   41.9136  
1      0.00  219.5820  
2      0.00    6.8714  
3      0.45 -383.0310  
4      0.20    2.5164  

[5 rows x 21 columns]

The code **data.shape** is retrieving the shape of the DataFrame data. The shape attribute of a DataFrame returns a tuple representing its dimensions in the form (number_of_rows, number_of_columns).

In [7]:

# Our data has 9994 rows and 21 coumns
data.shape


(9994, 21)

**data.columns** is retrieving the column labels (names) of the DataFrame data. It returns an Index object containing the column labels.

In [8]:
# 
data.columns


Index(['Row ID', 'Order ID', 'Order Date', 'Ship Date', 'Ship Mode',
       'Customer ID', 'Customer Name', 'Segment', 'Country', 'City', 'State',
       'Postal Code', 'Region', 'Product ID', 'Category', 'Sub-Category',
       'Product Name', 'Sales', 'Quantity', 'Discount', 'Profit'],
      dtype='object')



Let's explore the data by understanding the variable names of our data

**data["Segment"].unique()** is used to retrieve the unique values in the "Segment" column of the Dataset. It returns an array or Index containing all the unique values present in the specified column.

In [9]:

# there are three unique segments of sales; 'Consumer', 'Corporate', 'Home Office'
data["Segment"].unique()


array(['Consumer', 'Corporate', 'Home Office'], dtype=object)

In [10]:


# 'Second Class', 'Standard Class', 'First Class', 'Same Day' are the ship modes
data["Ship Mode"].unique()



array(['Second Class', 'Standard Class', 'First Class', 'Same Day'],
      dtype=object)

In [11]:


# Sales are made only in United States
data["Country"].unique()



array(['United States'], dtype=object)

The code **data["City"].nunique()** is used to count the number of unique values in the "City" column of the Dataset. The nunique() method returns the number of distinct (unique) values in a Series.

In [12]:


# Sales was made in 531 states in the United States
data["City"].nunique()



531

In [13]:


# There are four unique Regions in the dataset; 'South', 'West', 'Central', 'East'
print(data["Region"].nunique())
data["Region"].unique()



4


array(['South', 'West', 'Central', 'East'], dtype=object)

In [14]:


# 'Furniture', 'Office Supplies', 'Technology' are the categories of products for sale
print(data["Category"].nunique())
data["Category"].unique()



3


array(['Furniture', 'Office Supplies', 'Technology'], dtype=object)

In [15]:

# There are 17 unique Sub-Categories of products sold
print(data["Sub-Category"].nunique())
data["Sub-Category"].unique()



17


array(['Bookcases', 'Chairs', 'Labels', 'Tables', 'Storage',
       'Furnishings', 'Art', 'Phones', 'Binders', 'Appliances', 'Paper',
       'Accessories', 'Envelopes', 'Fasteners', 'Supplies', 'Machines',
       'Copiers'], dtype=object)

In [16]:

# 1850 different products were sold by the company
print(data["Product Name"].nunique())
data["Product Name"].unique()



1850


array(['Bush Somerset Collection Bookcase',
       'Hon Deluxe Fabric Upholstered Stacking Chairs, Rounded Back',
       'Self-Adhesive Address Labels for Typewriters by Universal', ...,
       'Eureka Hand Vacuum, Bagless', 'LG G2',
       'Eldon Jumbo ProFile Portable File Boxes Graphite/Black'],
      dtype=object)

**data.describe()** is used to generate descriptive statistics of the numerical columns in the DataFrame data. It provides summary statistics such as mean, standard deviation, minimum, 25th percentile (Q1), median (50th percentile or Q2), 75th percentile (Q3), and maximum for each numeric column.

In [17]:

# The descriptive statistics of 'Row ID' and 'Postal Code' is not needed
# And they would not be need in this analysis so we may drop or delete them

columns_to_drop = ['Row ID', 'Postal Code']
data = data.drop(columns=columns_to_drop)

data.describe()


Sales     Quantity     Discount       Profit
count   9994.000000  9994.000000  9994.000000  9994.000000
mean     229.858001     3.789574     0.156203    28.656896
std      623.245101     2.225110     0.206452   234.260108
min        0.444000     1.000000     0.000000 -6599.978000
25%       17.280000     2.000000     0.000000     1.728750
50%       54.490000     3.000000     0.200000     8.666500
75%      209.940000     5.000000     0.200000    29.364000
max    22638.480000    14.000000     0.800000  8399.976000

## **FEATURE ENGINEERING / DATA CLEANING**

The code below uses the **isna()** method to create a boolean mask where True represents NaN values, and then **sum()** is applied to count the number of True values (which are the NaN values) along each column.

If you want the total count of NaN values in the entire DataFrame, you can use **data.isna().sum().sum()**.

In [18]:


# There are no NaN values in the dataset
print(data.isna().sum())
print(data.isna().sum().sum())



Order ID         0
Order Date       0
Ship Date        0
Ship Mode        0
Customer ID      0
Customer Name    0
Segment          0
Country          0
City             0
State            0
Region           0
Product ID       0
Category         0
Sub-Category     0
Product Name     0
Sales            0
Quantity         0
Discount         0
Profit           0
dtype: int64
0


* `pd.to_datetime(data['Order Date'])`: This converts the 'Order Date' column in the DataFrame 'data' to datetime format. It's helpful when dealing with date-related operations, as it allows you to perform various time-based analyses.

* `pd.to_datetime(data['Ship Date'])`: Similarly, this line converts the 'Ship Date' column in the DataFrame 'data' to datetime format.

In [19]:

data['Order Date'] = pd.to_datetime(data['Order Date'])
data['Ship Date'] = pd.to_datetime(data['Ship Date']) 


* `data['Order Month'] = data['Order Date'].dt.month`: This line creates a new column 'Order Month' and assigns it the month  values extracted from the 'Order Date' column. The .dt.month accessor extracts the month component from each datetime entry.

* `data['Order Year'] = data['Order Date'].dt.year`: Similarly, this line creates a new column 'Order Year' and assigns it the year values extracted from the 'Order Date' column using the .dt.year accessor.

* `data['Order Day of Week'] = data['Order Date'].dt.dayofweek`: This line creates a new column 'Order Day of Week' and assigns it the day of the week values (0 to 6, where Monday is 0 and Sunday is 6) extracted from the 'Order Date' column using the .dt.dayofweek accessor.

In [20]:


data['Order Month'] = data['Order Date'].dt.month 
data['Order Year'] = data['Order Date'].dt.year
data['Order Day of Week'] = data['Order Date'].dt.dayofweek



## **ANALYSIS**



Let us conduct a comprehensive analysis of the sales data to gain insights into the dynamics related to products, regions, categories, and customer segments. This analysis aims to provide a nuanced understanding of areas for strategic focus and potential areas that require improvement or mitigation.

The code below is performing the following tasks:

1. It groups the data by the 'Order Month' column.
2. It sums up the 'Sales' values for each month.
3. It resets the index of the resulting DataFrame.
4. It uses Plotly Express (`px`) to create a line chart (`line` method).
5. The x-axis of the chart represents the 'Order Month,' and the y-axis represents the total 'Sales.'
6. The chart is given the title 'Monthly Sales Analysis.'
7. Finally, it displays the chart using `fig.show()`.

In [21]:


sales_by_month = data.groupby('Order Month')['Sales'].sum().reset_index()

fig = px.line(sales_by_month, 
              x='Order Month', 
              y='Sales', 
              title='Monthly Sales Analysis')
fig.show()


The analysis of sales volume indicates a consistent upward trend throughout the year with noticeable seasonal fluctuations. Noteworthy spikes are observed, particularly during mid-February to early spring, which aligns with the onset of warmer weather. Another significant surge occurs in August, followed by a dip in September, and a subsequent sharp rise in October leading to a peak in November. Although sales remain high through November, there is a gradual decline.

Historical data suggests a correlation between increased consumer spending on goods during the holiday season, driven by prominent shopping events like Prime Day, Cyber Monday, and Black Friday. In light of these patterns, it is crucial for the company to strategically capitalize on these seasonal spikes by acquiring and retaining customers.

To leverage these opportunities, the firm should implement targeted advertising campaigns aimed at attracting new customers during peak seasons and fostering customer retention throughout the entire year. This approach will not only enhance customer acquisition during high-demand periods but also contribute to sustained demand for the company's products year-round.


The code below is performing the following tasks:

* sales_by_category: It groups the DataFrame (data) by the 'Category' column and calculates the sum of sales for each category. The result is stored in a new DataFrame.

* fig: It creates a pie chart (`px.pie`) using the sales data (sales_by_category). The 'Category' column is used for the names of the pie slices, and the 'Sales' column is used for the values. The hole parameter creates a donut-like appearance.

* `fig.update_traces`: It updates the appearance of the pie chart by placing the text inside the slices (textposition='inside') and displaying both the percentage and label (textinfo='percent+label').

* `fig.update_layout`: It updates the layout of the chart by setting the title text to 'Sales Analysis by Category' and specifying the font size.

* `fig.show()`: It displays the pie chart.

In [22]:


sales_by_category = data.groupby('Category')['Sales'].sum().reset_index()


fig = px.pie(sales_by_category, 
             values='Sales', 
             names='Category', 
             hole=0.5, 
             color_discrete_sequence=px.colors.qualitative.Pastel)

fig.update_traces(textposition='inside', textinfo='percent+label')
fig.update_layout(title_text='Sales Analysis by Category', title_font=dict(size=24))

fig.show()


In [24]:
data.columns

Index(['Order ID', 'Order Date', 'Ship Date', 'Ship Mode', 'Customer ID',
       'Customer Name', 'Segment', 'Country', 'City', 'State', 'Region',
       'Product ID', 'Category', 'Sub-Category', 'Product Name', 'Sales',
       'Quantity', 'Discount', 'Profit', 'Order Month', 'Order Year',
       'Order Day of Week'],
      dtype='object')

In [26]:
sales_by_Quantity = data.groupby('Category')['Quantity'].sum().reset_index()
sales_by_Quantity

Category  Quantity
0        Furniture      8028
1  Office Supplies     22906
2       Technology      6939

Our analysis reveals that the Technology category dominates the sales performance, contributing 36.4% to the total sales. Following closely is Furniture, which accounts for 32.3% of the total sales, and Office Supplies comes next with 31.3%. Despite Technology products having the lowest sales quantity, their higher unit prices result in the highest sales revenue. The findings suggest an opportunity for the company to strategically focus and invest more in the Technology category, given its significant contribution to overall sales. While prioritizing Technology, it is essential to maintain attention on Furniture and Office Supplies, as all three categories play vital roles in the overall sales revenue.

The provided code below is creating a bar chart using Plotly Express to visualize the total sales for each sub-category.


1. `sales_by_subcategory`: It groups the DataFrame (`data`) by the 'Sub-Category' column and calculates the sum of sales for each sub-category. The result is stored in a new DataFrame.

2. `fig`: It creates a bar chart (`px.bar`) using the sales data (`sales_by_subcategory`). The 'Sub-Category' column is used for the x-axis (horizontal axis), and the 'Sales' column is used for the y-axis (vertical axis).

3. `title='Sales Analysis by Sub-Category'`: It sets the title of the bar chart to 'Sales Analysis by Sub-Category'.

4. `fig.show()`: It displays the bar chart.



In [13]:


sales_by_subcategory = data.groupby('Sub-Category')['Sales'].sum().reset_index()
fig = px.bar(sales_by_subcategory, 
             x='Sub-Category', 
             y='Sales', 
             title='Sales Analysis by Sub-Category')
fig.show()

In [27]:
sales_by_subcategory = data.groupby('Sub-Category')['Sales'].sum().reset_index()
fig = px.bar(sales_by_subcategory, 
             x='Sub-Category', 
             y='Sales', 
             title='Sales Analysis by Sub-Category',
             category_orders={"Sub-Category": sales_by_subcategory.sort_values('Sales')['Sub-Category'].tolist()})
fig.show()


In [28]:
sales_by_subcategory = data.groupby(['Category', 'Sub-Category'])['Sales'].sum().reset_index()
fig = px.bar(sales_by_subcategory, 
             x='Sub-Category', 
             y='Sales', 
             title='Sales Analysis by Sub-Category',
             color='Category',  # Add color parameter for legend
             category_orders={"Sub-Category": sales_by_subcategory.sort_values('Sales')['Sub-Category'].tolist()})
fig.show()


The analysis of sub-categories reveals notable trends in sales performance. Phones, a product within the technology category, emerged as the top performer with sales totaling $330,007. Following closely is the furniture category, specifically chairs, contributing $328,449 to the sales revenue. 

Conversely, products like fasteners, labels, envelopes, art supplies, and paper from the Office Supplies category yielded comparatively lower sales revenue. This is attributed to their lower unit prices. 

While these Office Supplies items may generate less revenue individually, they play a strategic role in attracting frequent visits from office staff. This foot traffic serves as a catalyst for lead generation, particularly for higher-priced items. Consequently, these leads translate into increased sales and contribute significantly to the overall sales revenue. It is advisable to recognize the dual impact of these seemingly lower-performing products, acknowledging their role in customer engagement and subsequent sales conversion.



The provided code below is creating a line chart using Plotly Express to visualize the total profit for each month.



1. `profit_by_month`: It groups the DataFrame (`data`) by the 'Order Month' column and calculates the sum of profits for each month. The result is stored in a new DataFrame.

2. `fig`: It creates a line chart (`px.line`) using the profit data (`profit_by_month`). The 'Order Month' column is used for the x-axis (horizontal axis), and the 'Profit' column is used for the y-axis (vertical axis).

3. `title='Monthly Profit Analysis'`: It sets the title of the line chart to 'Monthly Profit Analysis'.

4. `fig.show()`: It displays the line chart.



In [14]:


profit_by_month = data.groupby('Order Month')['Profit'].sum().reset_index()
fig = px.line(profit_by_month, 
              x='Order Month', 
              y='Profit', 
              title='Monthly Profit Analysis')
fig.show()



The analysis of profits mirrors the trend observed in sales revenue, displaying a consistent upward trajectory with noticeable spikes. The profit profile exhibits a sharp increase from February, followed by a slight dip in March. Subsequently, profits maintain an ascending trend from April to the year's end, with a brief decline noted in July. As anticipated, peak profits coincide with holiday periods.

It is evident that profit trends align closely with sales volume throughout the year. This synchronicity underscores the direct correlation between sales performance and overall profitability. The firm should strategically align its operational and marketing efforts to capitalize on these observed patterns, focusing on optimizing profits during peak seasons and sustaining profitability throughout the entire fiscal year.


Here's a breakdown of the code below;

1. `profit_by_category`: It groups the DataFrame (`data`) by the 'Category' column and calculates the sum of profits for each category. The result is stored in a new DataFrame.

2. `fig`: It creates a pie chart (`px.pie`) using the profit data (`profit_by_category`). The 'Profit' column is used as values, the 'Category' column is used as names, and the chart colors are set using the Pastel color sequence.

3. `hole=0.5`: It creates a donut chart with a hole in the center, where `hole=0.5` specifies the ratio of the hole size to the total chart size.

4. `fig.update_traces(textposition='inside', textinfo='percent+label')`: It adjusts the placement of text labels inside the pie slices, and it displays both the percentage and the label.

5. `fig.update_layout(title_text='Profit Analysis by Category', title_font=dict(size=24))`: It sets the title of the pie chart to 'Profit Analysis by Category' with a font size of 24.

6. `fig.show()`: It displays the pie chart.


In [15]:


profit_by_category = data.groupby('Category')['Profit'].sum().reset_index()

fig = px.pie(profit_by_category, 
             values='Profit', 
             names='Category', 
             hole=0.5, 
             color_discrete_sequence=px.colors.qualitative.Pastel)

fig.update_traces(textposition='inside', textinfo='percent+label')
fig.update_layout(title_text='Profit Analysis by Category', title_font=dict(size=24))

fig.show()



Upon analyzing profit distribution among categories, the pie chart reveals that 'Technology' accounts for a significant majority, contributing over half (50.8%) to the total profits. Following closely is the 'Office Supplies' category, contributing 42.8% to the overall profits.

Surprisingly, despite being the second-highest in terms of sales, the 'Furniture' category underperforms, contributing only 6.44% to the total profits. This discrepancy warrants further investigation into furniture sales, examining factors such as discounts offered and their impact on profit margins. The Superstore should implement measures to enhance sales in the Technology category while addressing challenges within the Furniture category.

A deeper examination reveals that the Furniture category experienced the highest discount, amounting to $123,516.54, compared to $76,404.94 from Office Supplies and $122,660.65 from Technology. This discounting strategy significantly impacted the profits derived from the Furniture category and should be a focal point for strategic adjustments.

In [33]:
data['Discount Amount'] = data['Discount']*data['Sales']

In [34]:
Discount_Amount = data.groupby('Category')['Discount Amount'].sum().reset_index()
Discount_Amount

Category  Discount Amount
0        Furniture    123516.541041
1  Office Supplies     76404.940600
2       Technology    122660.652700

 
 the code below generates a bar chart to visualize the profits for each sub-category. Each bar on the chart represents a sub-category, and the height of the bar corresponds to the total profit for that sub-category. Like we did for the Sales Analysis by Sub-Category early on.
 

In [16]:


profit_by_subcategory = data.groupby('Sub-Category')['Profit'].sum().reset_index()
fig = px.bar(profit_by_subcategory, x='Sub-Category', 
             y='Profit', 
             title='Profit Analysis by Sub-Category')
fig.show()



Upon examining profits by sub-categories, it is evident that Copiers and Phones contribute significantly to overall profits. In contrast, Fasteners are marginally profitable, while Tables and Bookcases are experiencing losses, displaying negative profits. Notably, Tables, despite generating one of the highest sales revenues, incur the most substantial losses.

Tables and Bookcases fall under the Furniture category, contributing to the overall lower profits observed in this category. Unless Tables and Bookcases serve as loss leaders with a strategic purpose, it is imperative for the company to implement measures to enhance their profitability and transform them into positive contributors to overall profits. This could involve reassessing pricing strategies, cost structures, or exploring avenues for operational efficiency.

The provided code below creates a grouped bar chart using Plotly to visualize both sales and profit data by customer segment. Here's a breakdown of the code:

1. `sales_profit_by_segment`: It groups the DataFrame (`data`) by the 'Segment' column and calculates the sum of both 'Sales' and 'Profit' for each segment. The result is stored in a new DataFrame.

2. `color_palette`: It defines a color palette for the bars in the chart using Plotly's qualitative color palette called 'Pastel'.

3. `fig`: It creates a bar chart (`go.Figure()`) with two traces (bars). The first trace represents 'Sales' (colored with `color_palette[0]`), and the second trace represents 'Profit' (colored with `color_palette[1]`). Each bar corresponds to a customer segment.

4. `fig.update_layout`: It updates the layout of the chart, setting the title, x-axis title ('Customer Segment'), and y-axis title ('Amount').

5. `fig.show()`: It displays the grouped bar chart with both sales and profit data.

In summary, the code generates a grouped bar chart that visually compares sales and profit amounts for different customer segments. Each customer segment has two bars (one for sales and one for profit) side by side. The title of the chart is 'Sales and Profit Analysis by Customer Segment'.

In [17]:


sales_profit_by_segment = data.groupby('Segment').agg({'Sales': 'sum', 'Profit': 'sum'}).reset_index()

color_palette = colors.qualitative.Pastel

fig = go.Figure()
fig.add_trace(go.Bar(x=sales_profit_by_segment['Segment'], 
                     y=sales_profit_by_segment['Sales'], 
                     name='Sales',
                     marker_color=color_palette[0]))
fig.add_trace(go.Bar(x=sales_profit_by_segment['Segment'], 
                     y=sales_profit_by_segment['Profit'], 
                     name='Profit',
                     marker_color=color_palette[1]))

fig.update_layout(title='Sales and Profit Analysis by Customer Segment',
                  xaxis_title='Customer Segment', yaxis_title='Amount')

fig.show()



We created segments for the Soperstore's customer base, i.e 'Consumer, Corporate and Home Office. the consumer segment of the customers recored the highest sales revenue and also the highest profits. The Corporate segment of the customer ase recorded the second highest sales revenue as well as the second highest sales revenue as well as the second highest profits. lastly, the Home Office segment recorded the sales revenues as well as the least profits. This means that higher sales revenues corresponds to higher profits. 

The code below calculates the sales-to-profit ratio for each customer segment in the 'data' DataFrame. Here's a breakdown of the code:

1. `sales_profit_by_segment`: It groups the DataFrame (`data`) by the 'Segment' column and calculates the sum of both 'Sales' and 'Profit' for each segment using the `agg` function. The result is stored in a new DataFrame.

2. `sales_profit_by_segment['Sales_to_Profit_Ratio']`: It adds a new column to the DataFrame, 'Sales_to_Profit_Ratio', which represents the ratio of 'Sales' to 'Profit' for each customer segment.

3. `print(sales_profit_by_segment[['Segment', 'Sales_to_Profit_Ratio']])`: It prints the selected columns ('Segment' and 'Sales_to_Profit_Ratio') of the DataFrame to the console.



In [18]:


# the code calculates the sales-to-profit ratio for each customer segment and 
# displays the corresponding values in a tabular format. This ratio provides insights into 
# how efficiently each segment is converting sales into profits.

sales_profit_by_segment = data.groupby('Segment').agg({'Sales': 'sum', 'Profit': 'sum'}).reset_index()
sales_profit_by_segment['Sales_to_Profit_Ratio'] = sales_profit_by_segment['Sales'] / sales_profit_by_segment['Profit']
print(sales_profit_by_segment[['Segment', 'Sales_to_Profit_Ratio']])



       Segment  Sales_to_Profit_Ratio
0     Consumer               8.659471
1    Corporate               7.677245
2  Home Office               7.125416


The Sales to Profit Ratio provides insight into the relationship between sales and profits for different customer segments. The ratio represents how many times the sales amount is compared to the profit amount. 

In this specific table:

- Consumer segment has a Sales to Profit Ratio of approximately 8.66, indicating that for every dollar of profit, there are 8.66 dollars in sales.
  
- Corporate segment has a ratio of around 7.68, implying that for every dollar of profit, there are 7.68 dollars in sales.

- Home Office segment has a ratio of about 7.13, suggesting that for every dollar of profit, there are 7.13 dollars in sales.

A higher ratio generally signifies that the segment is more efficient in converting sales into profits. It's essential for the company to analyze these ratios to understand the effectiveness of their strategies in different customer segments and consider adjustments if needed to improve overall profitability.

## **SUMMARY**

Based on the insights gleaned from the analysis, the following actions are recommended:

1. Implement targeted advertising campaigns: Capitalize on seasonal spikes, especially during holiday shopping events, by launching strategic advertising campaigns to attract new customers and foster retention throughout the year.

2. Strategic investment in Technology: Recognize the dominance of the Technology category and strategically invest resources to maximize returns. Focus on maintaining a balance with Furniture and Office Supplies for a comprehensive revenue approach.

3. Investigate and adjust Furniture sales strategy: Conduct a thorough investigation into Furniture category sales, particularly examining the impact of discounts. Implement strategic adjustments to enhance sales performance and overall profitability.

4. Optimize sub-category profitability: Identify sub-categories with significant contributions to profits, such as Copiers and Phones. Develop strategies to enhance profitability in marginally profitable categories and transform loss-incuring sub-categories, like Tables, into positive contributors.

5. Analyze and adjust Sales to Profit Ratios: Regularly analyze Sales to Profit Ratios for different customer segments (Consumer, Corporate, Home Office) to understand efficiency in converting sales into profits. Consider strategic adjustments to improve overall profitability based on these insights.